backend/app/services/sem_search.py

```
@@ -40,11 +67,26 @@ def search_unstructured(
+    use_expansion = getattr(settings, "ENABLE_CLIP_EXPANSION", True) and expanded_queries and len(expanded_queries) > 1
+    if use_expansion:
+        try:
+            embedder = get_embedder()
+            embs = embedder.text_embed(expanded_queries)
+            n = len(embs)
+            w_primary = 0.6
+            w_rest = (1.0 - w_primary) / max(1, n - 1)
+            weights = [w_primary] + [w_rest] * (n - 1)
+            combined = np.average(embs, axis=0, weights=weights).astype(np.float32)
+            query_emb = combined
+        except Exception:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)